### PR TITLE
fix(sort-modules): fix `type: 'usage'` not handling `fallbackSort`

### PR DESCRIPTION
**Pull request by hugop95**
- Related issue: https://github.com/azat-io/eslint-plugin-perfectionist/issues/666

## Description

`type: 'usage'` currently doesn't handle `fallbackSort` correctly, because this comparator never returns `0`. This PR fixes this issue.

It also improves the documentation for this option to reduce confusion.

## Important note

There has been issues with dependency sorting before, and I'm wondering if I can't find a counter-example where the current implementation in this PR actually doesn't work. I want to spend more time investigating this before opening.

Edit: See https://github.com/azat-io/eslint-plugin-perfectionist/pull/667#discussion_r2669701543.

## What is the purpose of this pull request?

- [x] Bug fix

### CODE REVIEW

**Comment by azat-io** [comment]
I would like to create a release, we had an important bug fix:
https://github.com/azat-io/eslint-plugin-perfectionist/pull/670

At the same time, this pull request also seems important to me. Please advise, should I create a release today or wait for this pull request?

**Comment by hugop95** [comment]
@azat-io 

> At the same time, this pull request also seems important to me. Please advise, should I create a release today or wait for this pull request?

Please don't wait for this PR, I still need to test additional things as I suspect there might be edge cases that are not covered.

File: rules/sort-modules/build-usage-comparator.ts [hunk 1]
```diff
@@ -26,28 +34,63 @@ export function buildUsageComparator({
...
+      options.order,
+    )
+
+    if (sortedOrderedValue !== unsortedOrderedValue) {
+      return sortedOrderedValue
+    }
+
+    let aWithUpdatedDependencies = updatedSortingNodeByNode.get(nodeA)!
+    let bWithUpdatedDependencies = updatedSortingNodeByNode.get(nodeB)!
+    if (
```
Comment (hugop95) [inline rules/sort-modules/build-usage-comparator.ts:66]: @reviewers

A part of me is not convinced that this will catch all cases, but I couldn't find a counter-example.

Dependency sorting is necessary topological, and we can't replicate it perfectly just with JavaScript's `sort`. This is why sorting by dependencies is always done at the end and [isn't based on `sort`](https://github.com/azat-io/eslint-plugin-perfectionist/blob/271a4811f1cb0f6ca3f4fafc64bf23f87c305516/utils/sort-nodes-by-dependencies.ts).

Here, we add a `fallback` sort after sorting by usage (which is essentially sorting by dependencies), so I'm afraid that there might be edge cases with transitive dependencies between nodes that might lead to issues.

I think it's acceptable to go with the current implementation and see where it goes, as it's quite a niche use case. The only way to really know if this algorithm works 100% of the time is to see if somebody can make it fail and reports it.

Comment (hugop95) [reply rules/sort-modules/build-usage-comparator.ts:66]: @coderabbitai

**Comment by azat-io** [comment]
Thanks!
